flink-catalog-aws-glue/src/main/java/org/apache/flink/table/catalog/glue/util/AwsClientFactories.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog.glue.util;
+
+import org.apache.flink.configuration.ReadableConfig;
+
+import software.amazon.awssdk.services.glue.GlueClient;
+
+/** Default factories. */
+public class AwsClientFactories {
+
+    private AwsClientFactories() {}
+
+    public static AwsClientFactory factory(AwsProperties properties) {
+        return new DefaultAwsClientFactory(properties);
+    }
+
+    static class DefaultAwsClientFactory implements AwsClientFactory {
+
+        /** instance that holds provides. */
+        private AwsProperties awsProperties;
+
+        DefaultAwsClientFactory(AwsProperties properties) {
+            awsProperties = properties;
+        }
+
+        @Override
+        public GlueClient glue() {
+            return GlueClient.builder()
```

Comment:
How about credential providers? Can we use the util classes [#1](https://github.com/apache/flink-connector-aws/blob/main/flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSClientUtil.java), [#2](https://github.com/apache/flink-connector-aws/blob/main/flink-connector-aws-base/src/main/java/org/apache/flink/connector/aws/util/AWSGeneralUtil.java) to build clients and credential providers so the configuration/options/features is consistent/reused for all integrations?